packages/lib/services/search/gotoAnythingStyleQuery.ts

```diff
@@ -1,10 +1,9 @@
 import { isCallbackUrl } from '../../callbackUrlUtils';
-import isItemId from '../../models/utils/isItemId';
 
 export default (query: string) => {
 	if (!query) return '';
 
-	if (isItemId(query) || isCallbackUrl(query)) return query;
+	if (isCallbackUrl(query)) return query;
```

Comment:
Thank you for working on this!

I'm still getting the error if I search for `joplin://x-callback-url/openNote?`. Locally, it seems to be related to `GotoAnything.tsx` assuming that search terms are regular expressions on this line:

https://github.com/laurent22/joplin/blob/bcff7ac1d3aaecedc627927bad336cd02f473003/packages/app-desktop/plugins/GotoAnything.tsx#L380

If the `term` has [`.type === 'text'`](https://github.com/laurent22/joplin/blob/546c75b9558345f34b11729334e45752a438b2a1/packages/lib/services/search/SearchEngine.ts#L58), then `valueRegex` can be `undefined`. This is the case, for example, when a search term parsed with `parseQuery` does not contain a `*`:
https://github.com/laurent22/joplin/blob/546c75b9558345f34b11729334e45752a438b2a1/packages/lib/services/search/SearchEngine.ts#L568-L571

Based on this, another possible fix could be to handle the case where a keyword has `.type === 'text'` (and `.valueRegex === undefined`).

Reply:
It seems the error occurs because `valueRegex` is not a member of the `Term` interface. I updated it to `value`, and it works fine now.

```typescript
export interface Term {
    name: string;
    value: string;
    negated: boolean;
    quoted?: boolean;
    wildcard?: boolean;
}
```